requires.py

```diff
@@ -54,3 +59,14 @@ def get_master_info(self):
             'master': conv.get_remote('master'),
         }
         return data
+
+    def get_master_url(self):
+        conv = self.conversation()
+        return conv.get_remote('connection_string')
+
+    def get_master_ip(self):
+        conv = self.conversation()
+        return conv.get_remote('master')
+
+    def is_scaled(self):
+        return len(self.conversation().units) > 1
```

Comment:
That, and actually `is_scaled`, were artifacts of the "subordinate to spark" approach and aren't used now.  The "this node is master" doesn't even make sense in the non-subordinate case because the relation may represent multiple remote units.

On the flip side, the `is_scaled` check here wouldn't make sense or work in the subordinate case, because I think it would only see the single unit it's attached to.  I should probably remove this as well.